app.js

```diff
@@ -1,26 +1,43 @@
-const value = document.getElementById('passwordLength');
-const password = document.getElementById('returnedPassword')
-const badPassword = document.getElementById('badPassword')
-
-createPassword = () => {
-
-    let base = "abcdefghijklmnopqrstuvwxyz0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ!@#$%^&*";
-    let passwordLength = value.value;
-    let returnedPassword = "";
-
-    if (passwordLength <= 5 || passwordLength >= 41) {
-        badPassword.innerHTML = "Your password need to be between 6 to 40 characters"
-        password.innerHTML = ""
-    }else{
-        badPassword.innerHTML = ""
-        for (let i = 0, charLength = base.length; i < passwordLength; i++) {
-            returnedPassword += base.charAt(Math.floor(Math.random() * charLength))
-        }
-        return password.innerHTML = `Your password: ${returnedPassword}`
-    }
+let numbers = "0123456789";
+let lowercase = "abcdefghijklmnopqrstuvwxyz";
+let uppercase = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+let symbols = "!@#$%^&*";
 
+const returnedPassword = document.querySelector('.returnedPassword');
+
+createPassword = () =>{
+
+    
+    const form = document.querySelector('#inputForm');
+    const formData = new FormData(form);
+    const searchParams = new URLSearchParams(formData);
+    const passwordLength = document.querySelector('select').value;
+
+    const formResults = [...searchParams.entries()].reduce(
+        (result, [key, value]) => ({
+          ...result,
+          [key]: value,
+        }), {}
+      );
+
+    let base = [];
+
+    base = [
+        formResults.numbers ? base + numbers : '',
+        formResults.lowercase ? base + lowercase : '',
+        formResults.uppercase ? base + uppercase : '',
+        formResults.symbols ? base + symbols : ''
+    ]
+    const filledBase = base[0]+base[1]+base[2]+base[3];
```

Comment:
jeśli zamiast 
```js
formResults.numbers ? base + numbers : '',
```
skorzystasz z 
```js
...(formResults.numbers ? base + numbers : []),
```

to dostaniesz listę znaków w formie `base = ['a', 'b', 'c', '1', '@']`

wtedy nie będziesz musiał łączyć tych elementów w `filledBase` niepotrzebnie konwertując na stringa.